eevee/metrics/classification.py

```diff
@@ -98,55 +103,103 @@ def intent_report(
+        "layer": LAYER_PREFIX.format(label),
+        "precision": p,
+        "recall": r,
+        "f1-score": f,
+        "support": support
+    }
+    
+    return wgin
+
+
 def intent_layers_report(
         true_labels: pd.DataFrame,
         pred_labels: pd.DataFrame,
         intent_layers: Optional[Dict[str, Dict[str, List[str]]]] = None,
         breakdown=False,
 ):
+
     df = pd.merge(true_labels, pred_labels, on="id", how="inner")
 
     # for cases where we are seeing NaN values popping up.
-    df[['intent_x', 'intent_y']] = df[['intent_x', 'intent_y']].fillna(value="_")
+    df[[TRUE, PREDICTED]] = df[[TRUE, PREDICTED]].fillna(value="_")
 
-    # aliasing preds
-    col = "intent_y"
+    # aliasing predicted column with values provided
+    col = PREDICTED
     intents_dict = {value: key for key, values in intent_layers.get(col).items() for value in values}
     df[col] = df[col].apply(lambda intent: intents_dict.get(intent, intent))
 
-    #aliasing trues
-    col = "intent_x"
+    #aliasing true column with values provided.
+    col = TRUE
     intents_dict = {value: key for key, values in intent_layers.get(col).items() for value in values}
-    df["{}-alias".format(col)] = df[col].apply(lambda intent: intents_dict.get(intent, intent))
+    df[ALIAS_SUFFIX.format(col)] = df[col].apply(lambda intent: intents_dict.get(intent, intent))
+
+    # first element is taken as the name of the original layer
+    PREDICTED_LAYER = list(intent_layers.get(PREDICTED).keys())[0]
 
-    PREDICTED_LAYER = list(intent_layers.get("intent_y").keys())[0]
+    # reverse aliasing dictionary - maps sublayers to original layer
+    REVERSE_OOS_DICT = {sub_layer: PREDICTED_LAYER for sub_layer in intent_layers.get(TRUE)}
 
     # where each intent group is having its own classification_report
     if breakdown:
 
         return_output_as_dict = True
         grouped_classification_reports = {}
 
-        for sub_layer in intent_layers.get("intent_x"):
-            col = "intent_y"
-            df["{}-alias".format(col)] = df[col].apply(lambda intent: {PREDICTED_LAYER: sub_layer}.get(intent, intent))
-            group_classification_report = classification_report(df["{}-alias".format("intent_x")],
-                                                                df["{}-alias".format("intent_y")],
-                                                                labels=[sub_layer], output_dict=return_output_as_dict, zero_division=0)
-            group_classification_report_df = pd.DataFrame(group_classification_report).transpose()
-            group_classification_report_df["support"] = group_classification_report_df["support"].astype('int32')
-            grouped_classification_reports["layer-{}".format(sub_layer)] = group_classification_report_df
+        for sub_layer in intent_layers.get(TRUE):
+
+            col = PREDICTED
+            df[ALIAS_SUFFIX.format(col)] = df[col].apply(lambda intent: {PREDICTED_LAYER: sub_layer}.get(intent, intent))
```

Comment:
i find these two lines repeating a lot and confusing. this is the only thing holding me back.

`PREDICTED_LAYER` is a `str` (constant throughout), `sub_layer` is also a `str`. then why the dict of 
`{PREDICTED_LAYER: sub_layer}.get(intent, intent)` ?

is this what you are trying to do?
```
lambda intent: sub_layer if PREDICTED_LAYER == intent else intent
```

Reply:
**Short answer**, yes that what i am doing

**Long answer**. _why_ i am using a dictionary here, is to maintain regularity with what i do above this piece of code (this holds for wherever i used this piece of code)

``` python
df[ALIAS_SUFFIX.format(col)] = df[col].apply(lambda intent: intents_dict.get(intent, intent))
```

This line, and the one you referenced go together. Below, i'll try to explain why - for ease of reference, let's call the above `line 1` and the one you referenced, `line 2`.

The goal of this PR is to calculate some numbers for sub-layers of a specific layer (for eg. acoustic and lexical are sub layers of the OOS layer). The assumption here, and this is what makes this PR necessary, is that only the `TRUE` tags contain the sub-layer intents, while the `PREDICTED` tags only contain the layer intent(s). 
Here, line 1 acts on the `TRUE` tags of the data, and aliases intents to the respective sub-layers. Now for each sub-layer, we want to calculate metrics. So `line 2` aliases the layer intent(s) in `PREDICTED` tags to the selected sub-layer. Keeping it a dictionary in `line 2` is to, both:

1. maintain regularity across pieces of code that do the same thing (in this case, aliasing)
2. allow flexibility for multiple intents in the layer, in the future